File: tests/src/com.google.protobuf/protobuf-javalite/3.25.5/src/test/java/com_google_protobuf/protobuf_javalite/SchemaUtilTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_google_protobuf.protobuf_javalite;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.junit.jupiter.api.Assertions.assertNotNull;

import com.google.protobuf.MapEntryLite;
import com.google.protobuf.SchemaUtilAccess;
import com.google.protobuf.WireFormat;
import org.junit.jupiter.api.Test;

public class SchemaUtilTest {
    @Test
    public void optionalFullUnknownFieldSchemaIsLoadedWhenAvailable() {
        Object fullSchema = SchemaUtilAccess.unknownFieldSetFullSchema();

        assertNotNull(fullSchema);
    }

    @Test
    public void mapDefaultEntryHolderIsResolvedFromMessageClassAndFieldName() {
        assertDoesNotThrow(() -> SchemaUtilAccess.mapDefaultEntry(MapDefaultEntryMessage.class, "labels"));
    }

    private static final class MapDefaultEntryMessage {
        private interface LabelsDefaultEntryHolder {
            MapEntryLite<String, Integer> defaultEntry =
                    MapEntryLite.newDefaultInstance(
                            WireFormat.FieldType.STRING, "", WireFormat.FieldType.INT32, 0);
        }
    }
}

File: tests/src/org.apache.pekko/pekko-protobuf-v3_3/1.0.1/src/test/scala/org_apache_pekko/pekko_protobuf_v3_3/SchemaUtilTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_pekko.pekko_protobuf_v3_3

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertInstanceOf
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test

class SchemaUtilTest {
  @Test
  def resolvesMapDefaultEntryHolderWhileBuildingGeneratedMessageSchema(): Unit = {
    assertNotNull(SchemaUtilMapFieldHost.initializedMapDefaultEntryForTests())

    try {
      SchemaUtilMapFieldHost.newMutable().parseEmptyInputWithGeneratedMessageSchema()
    } catch {
      case thrown: NullPointerException => assertEquals("mapDefaultEntry", thrown.getMessage)
      case thrown: RuntimeException => assertInstanceOf(classOf[NullPointerException], thrown.getCause)
    }
  }
}
